sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeFormatterHelper.scala

```diff
@@ -105,9 +105,9 @@ trait DateTimeFormatterHelper {
   // between legacy parser and new parser. If new parser fails but legacy parser works, throw a
   // SparkUpgradeException. On the contrary, if the legacy policy set to CORRECTED,
   // DateTimeParseException will address by the caller side.
-  protected def checkDiffResult[T](
+  protected def checkDiffParserResult[T](
       s: String, legacyParseFunc: String => T): PartialFunction[Throwable, T] = {
-    case e: DateTimeException if SQLConf.get.legacyTimeParserPolicy == EXCEPTION =>
+    case e if needConvertToSparkUpgradeException(e) =>
```

Comment:
The parser case can get`ArrayIndexOutOfBoundsException`, too?

Reply:
There could be such a risk for the parser to get it after roughly checked the JCL.

Reply:
Maybe RuntimeException is more appropriate